AIFoV/src/main/java/dk/gruppeseks/aifov/AIFoVProvider.java

```diff
@@ -0,0 +1,164 @@
+    {
+        List<Position> path = new ArrayList<>();
+        _world = world;
+        Position targetPos = rayCast(entity, target); // Providing the point on the circle it collided would result in better AI.
+
+        if (targetPos == null)
+        {
+
+        }
+        else
+        {
+            Vector2 distance = new Vector2(entity.get(Position.class), target.get(Position.class));
+            if (distance.getMagnitude() < DUMB_AI_RANGE)
+            {
+                path.add(new Position(target.get(Position.class))); // doesnt work as intended
+            }
+            else
+            {
+                path.add(targetPos);
+            }
+        }
+
+        Position[] pathArray = new Position[path.size()];
+        for (int i = 0; i < path.size(); i++)
+        {
+            pathArray[i] = path.get(i);
+        }
+        return new Path(pathArray);
+    }
+
+    private Position rayCast(Entity zombie, Entity target)
+    {
+        ArrayList<Float> shape = new ArrayList<>();
+        double rotSteps = VISION_DEGREES / (double)RAY_COUNT; // 1.5
+        Vector2 orientation = zombie.get(Body.class).getOrientation();
+        if (orientation == null)
+        {
+            orientation = new Vector2(0, 1);
+        }
+        Vector2 startRay = new Vector2(orientation);
+        startRay.rotateDegrees(-VISION_DEGREES / 2);
+
+        Position targetPos = target.get(Position.class);
+        Body targetBod = target.get(Body.class);
+        Position targetCenter = new Position(targetPos.getX() + targetBod.getWidth() / 2, targetPos.getY() + targetBod.getHeight() / 2);
+        Position p = zombie.get(Position.class);
+        Body b = zombie.get(Body.class);
+
+        Position zombieCenter = new Position(p.getX() + b.getWidth() / 2, p.getY() + b.getHeight() / 2);
+
+        boolean playerCollision = false;
+        for (int i = 0; i < RAY_COUNT; i++)
+        {
+            double rotAmount = rotSteps * i;
+            Vector2 ray = new Vector2(startRay);
+            ray.setMagnitude(VISION_DISTANCE);
+            ray.rotateDegrees(rotAmount);
+            Position collision = getFirstCollision(ray, zombie.get(Position.class));
+            if (!playerCollision && lineCircleCollision(targetCenter, target.get(Body.class), zombieCenter, collision))
+            {
+                playerCollision = true;
+            }
+            shape.add((float)collision.getX());
+            shape.add((float)collision.getY());
+        }
+
+        shape.add((float)zombieCenter.getX());
+        shape.add((float)zombieCenter.getY());
+
+        float[] floatShape = new float[shape.size()];
+        for (int i = 0; i < shape.size(); i++)
+        {
+            floatShape[i] = shape.get(i);
+        }
+        AIData aiData = zombie.get(AIData.class);
+        aiData.setFoVShape(floatShape);
+        if (playerCollision)
+        {
+            return targetCenter;
+        }
+        return null;
```

Comment:
because we need to draw the FoVShape eitherway, so the return null would be after aiData.setFoVShape(), and then it wouldnt really be any different than what it is currently.